src/main/java/io/camunda/testing/assertions/JobAssert.java

```diff
@@ -0,0 +1,105 @@
+package io.camunda.testing.assertions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.MapAssert;
+import org.assertj.core.data.Offset;
+
+// TODO discuss name: job assertions, service task assertions, worker assertions
+
+/** Assertions for {@code ActivatedJob} instances */
+public class JobAssert extends AbstractAssert<JobAssert, ActivatedJob> {
+
+  public JobAssert(final ActivatedJob actual) {
+    super(actual, JobAssert.class);
+  }
+
+  /**
+   * Asserts that the activated job is associated to an element with the given id
+   *
+   * @param expectedElementId element id to check
+   * @return this {@link JobAssert}
+   */
+  public JobAssert hasElementId(final String expectedElementId) {
+    assertThat(expectedElementId).describedAs("expectedElementId").isNotNull().isNotEmpty();
+    final String actualElementId = actual.getElementId();
+
+    assertThat(actualElementId)
+        .overridingErrorMessage(
```

Comment:
❌ In the `ProcessInstanceAssertions` I went with `withFailMessage`. They both do exactly the same but I think it would be nice to be consistent with one of the two. Since we decided our `UnhappyPath` tests are called `TestingSomethingFailure` my proposal would be to change these to `withFailMessage`.